File: app/src/main/java/com/github/android/lvrn/lvrnproject/view/adapter/datapostset/impl/TagsListAdapter.java
package com.github.android.lvrn.lvrnproject.view.adapter.datapostset.impl;


import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.recyclerview.widget.RecyclerView;

import com.github.android.lvrn.lvrnproject.databinding.ItemTagBinding;
import com.github.valhallalabs.laverna.persistent.entity.Tag;

import java.util.List;

/**
 * @author Andrii Bei <psihey1@gmail.com>
 */
//TODO: FIX!
public class TagsListAdapter extends RecyclerView.Adapter<TagsListAdapter.TagEditingViewHolder> {
    private List<Tag> mDataSet;

    public TagsListAdapter(List<Tag> data) {
        mDataSet = data;
    }

    @Override
    public TagEditingViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new TagEditingViewHolder(ItemTagBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false));
    }

    //TODO change after implements DAO
    @Override
    public void onBindViewHolder(TagEditingViewHolder holder, int position) {
        holder.itemTagBinding.checkedTextViewTags.setText(mDataSet.get(position).getName());
    }

    @Override
    public int getItemCount() {
        return mDataSet.size();
    }

    class TagEditingViewHolder extends RecyclerView.ViewHolder {

        ItemTagBinding itemTagBinding;

        TagEditingViewHolder(ItemTagBinding itemTagBinding) {
            super(itemTagBinding.getRoot());
            this.itemTagBinding = itemTagBinding;
        }
    }

}

File: app/src/main/java/com/github/android/lvrn/lvrnproject/view/adapter/datapostset/impl/TagsListAdapter.kt
package com.github.android.lvrn.lvrnproject.view.adapter.datapostset.impl

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.github.android.lvrn.lvrnproject.databinding.ItemTagBinding
import com.github.valhallalabs.laverna.persistent.entity.Tag

//TODO: FIX!
class TagsListAdapter(private val mDataSet : List<Tag>) : RecyclerView.Adapter<TagsListAdapter.TagEditingViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TagEditingViewHolder {
        return TagEditingViewHolder(
            ItemTagBinding.inflate(
                LayoutInflater.from(parent.context),
                parent,
                false
            )
        )
    }

    //TODO change after implements DAO
    override fun onBindViewHolder(holder: TagEditingViewHolder, position: Int) {
        holder.itemTagBinding.checkedTextViewTags.text = mDataSet[position].name
    }

    override fun getItemCount(): Int {
        return mDataSet.size
    }

    class TagEditingViewHolder(var itemTagBinding: ItemTagBinding) :
        RecyclerView.ViewHolder(itemTagBinding.getRoot()) {
    }
}